homeassistant/components/recorder/statistics.py

```diff
@@ -1492,9 +1490,9 @@ def statistic_during_period(
     state_unit = unit = metadata[1]["unit_of_measurement"]
     if state := hass.states.get(statistic_id):
         state_unit = state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
-    convert = _get_statistic_to_display_unit_converter(unit, state_unit, units)
-
-    return {key: convert(value) if convert else value for key, value in result.items()}
```

Comment:
I'll pull this out into another PR as I can fully manually test this without waiting for the frontend issue

Reply:
The other change likely isn't as useful on its own anyways since I couldn't successfully reduce the stack depth as we need the None check.  But I found a regression in the process so still a win

Reply:
We could get rid of the none check if I build another converter that handles None cases.  It might be a lot of dupe code though so may not be worth the maintenance burden. Will need to dig in more

Reply:
It's the same argument every time. We can make it a lru conversion factory